webapp/src/components/MessageList.tsx

```diff
@@ -0,0 +1,58 @@
+'use client';
+
+import ListItem from '@mui/material/ListItem';
+import { FC, useCallback } from 'react';
+import { FixedSizeList, ListChildComponentProps } from 'react-window';
+
+import MessageForm from '@/components/MessageForm';
+import { MessageData } from '@/utils/adapters';
+
+type MessageListProps = {
+  languageName: string;
+  messages: MessageData[];
+  projectName: string;
+  translations: Record<string, string>;
+};
+
+const MessageList: FC<MessageListProps> = ({
+  languageName,
+  messages,
+  projectName,
+  translations,
+}) => {
+  const renderRow = useCallback(
+    (props: ListChildComponentProps): JSX.Element => {
+      const { index, style } = props;
+      const message = messages[index];
+      return (
+        <ListItem key={index} component="div" disablePadding style={style}>
+          <MessageForm
+            languageName={languageName}
+            message={message}
+            projectName={projectName}
+            translation={translations[message.id] || ''}
+          />
+        </ListItem>
+      );
+    },
+    [languageName, messages, projectName, translations],
+  );
+
+  if (typeof window === 'undefined') {
+    return;
+  }
+
+  return (
+    <FixedSizeList
+      height={window.innerHeight}
```

Comment:
`MessageList` will not be re-rendered if `window.innerHeight`. The list's size can get out of sync with the window if we resize the window. Let's create an issue for that rater than delaying this merge further.

Reply:
Fixed in https://github.com/zetkin/lyra/pull/107/commits/09c5969640e72da38048e7a8a61440a3dbe9bfcc